pkg/core/region.go

```diff
@@ -71,8 +71,36 @@ type RegionInfo struct {
 	queryStats        *pdpb.QueryStats
 	flowRoundDivisor  uint64
 	// buckets is not thread unsafe, it should be accessed by the request `report buckets` with greater version.
-	buckets       unsafe.Pointer
-	fromHeartbeat bool
+	buckets unsafe.Pointer
+	// source is used to indicate region's source, such as FromHeartbeat/FromSync/FromStorage.
+	source RegionSource
+}
+
+// RegionSource is the source of region.
+type RegionSource uint32
+
+const (
+	// FromStorage means this region's meta info might be stale.
+	FromStorage RegionSource = iota
+	// FromSync means this region's meta info might be stale.
+	FromSync
+	// FromHeartbeat means this region's meta info is relatively fresher.
+	FromHeartbeat
+)
+
+// IsSourceStale means this region's meta info might be stale.
+func (r *RegionInfo) IsSourceStale() bool {
+	return r.source == FromStorage || r.source == FromSync
```

Comment:
why sync is stale? I think the follower can start coordinator immediately if it become leader(region info is from sync.).

Reply:
fixed